.github/actions/clip-install-dep/action.yml

```diff
@@ -0,0 +1,76 @@
+name: "Install clip Dependency"
+description: "Install clip Dependency using cache when possible"
+inputs:
+  cpu:
+    description: "CPU architecture to build for"
+    required: false
+    default: "x86_64"
+  version:
+    description: "Version of clip to build"
+    required: true
+  global_cache_index:
+    description: "Global cache index"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Check required inputs
+      shell: bash
+      run: |
+        [[ "${{ inputs.version }}" ]] || { echo "version input is empty" ; exit 1; }
+        [[ "${{ inputs.global_cache_index }}" ]] || { echo "global_cache_index input is empty" ; exit 1; }
+
+    - name: Cache clip
+      id: cache-clip
+      uses: actions/cache/restore@v4
+      with:
+        path: dependencies/clip_install
+        key: clip-${{inputs.version}}-${{runner.os}}-${{inputs.cpu}}-${{inputs.global_cache_index}}-0
```

Comment:
please use a hard-coded version for now, do not use inputs.version, it cannot work like that.